orm/iterator.go

```diff
@@ -92,6 +95,91 @@ func First(it Iterator, dest Persistent) (RowID, error) {
 	return binKey, nil
 }
 
+func Paginate(
+	it Iterator,
+	pageRequest *query.PageRequest,
+	dest ModelSlicePtr,
+) (*query.PageResponse, error) {
+	// if the PageRequest is nil, use default PageRequest
+	if pageRequest == nil {
+		pageRequest = &query.PageRequest{}
+	}
+
+	offset := pageRequest.Offset
+	key := pageRequest.Key
+	limit := pageRequest.Limit
+	countTotal := pageRequest.CountTotal
+
+	if offset > 0 && key != nil {
+		return nil, fmt.Errorf("invalid request, either offset or key is expected, got both")
+	}
+
+	if limit == 0 {
+		limit = 100
+
+		// count total results when the limit is zero/not supplied
+		countTotal = true
+	}
+
+	if it == nil {
+		return nil, errors.Wrap(ErrArgument, "iterator must not be nil")
+	}
+	defer it.Close()
+
+	var slice, t reflect.Value
+	typ, err := assertDest(dest, &slice, &t)
+	if err != nil {
+		return nil, err
+	}
+
+	end := offset + limit
+
+	var count uint64
+	var nextKey []byte
+	for {
+		obj := reflect.New(typ)
+		val := obj.Elem()
+		model := obj
+		if typ.Kind() == reflect.Ptr {
+			val.Set(reflect.New(typ.Elem()))
+			model = val
+		}
+
+		binKey, err := it.LoadNext(model.Interface().(codec.ProtoMarshaler))
+		if ErrIteratorDone.Is(err) {
+			slice.Set(t)
+			break
+		}
+
+		count++
+
+		if count <= offset {
+			continue
+		}
+
+		if err != nil {
+			return nil, err
+		}
```

Comment:
so we are ignoring the errors before the offset, but we are not ignoring the errors after `limit` if `countTotal` is set. Personally, I would not ignore any error.

Reply:
good point, fixed.